nix/test/rebuild/rebuild.nix

```diff
@@ -0,0 +1,331 @@
+{ pkgs, lib, ... }:
+let
+  node1ip = "192.168.0.1";
+  node2ip = "192.168.0.2";
+  nexus_uuid = "19b98ac8-c1ea-11ea-8e3b-d74f5d324a22";
+  replica1_uuid = "9a9843db-f715-4f52-8aa4-119f5df3d05d";
+  replica2_uuid = "9a9843db-f715-4f52-8aa4-119f5df3d06e";
+in
+{
+  name = "rebuild";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ paulyoong ];
+  };
+
+  nodes = {
+    node1 = { config, lib, ... }: {
```

Comment:
could you have a single node expression and pass the yaml as an input?

Reply:
Maybe, but there are a number of node specific settings there also, so I'm not sure that would be appropriate. 

Having looked at a number of tests in the NixOS repo (e.g. https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/3proxy.nix) the convention seems to be to specify each node configuration. Also the NixOS example seemed to suggest specifying individual machine configurations (https://nixos.org/nixos/manual/index.html#sec-writing-nixos-tests)

Reply:
The common/default stuff is part of what I'm looking at, hopefully I'll be able to get it generic enough to be useful here too, whilst still actually being a useful way of avoiding duplicate code.